### ISSUE TITLE
Why do I got different results?

### ISSUE THREAD

**Issue by QiChen2014** [biolab/orange3#2981]
##### Orange version
Orange3.11.0

##### Expected behavior
Get more info from Data Sampler widget; How can I get the same results between DataSampler and non-DataSampler?

##### Actual behavior
just return the parameters that I choosed; 

##### Steps to reproduce the behavior
1. Use Data Sampler to sample a dataset, parameters like this:
    Sampling type: Random sample with 70 % of data, stratified (if possible), deterministic
2. Train a logistc regression model and test it with test data on Test&Score , then save the results;
3. Extract the trainning data and test data from Data Sampler, train and test a same logistic regression model with thes data respectively, then compare the results with the prior results.

##### Additional info (worksheets, data, screenshots, ...)
I sampled a subset from the input dataset by Data Sampler widget to train a Logistic Regression , parameters like this:

Sampling type: Random sample with 70 % of data, stratified (if possible), deterministic
Input: 178 instances
Sample: 125 instances
Remaining: 53 instances

then I used the Test&Score widget to see the result by setting "Stratified 10-fold Cross validation" and I got the result like this:

Method AUC CA F1 Precision Recall
Logistic Regression 0.751 0.776 0.757 0.756 0.776

the test result on testdata like this:

Method AUC CA F1 Precision Recall
Logistic Regression 0.709 0.736 0.709 0.704 0.736

After that, I extracted the trainning data and test data which sampled by the Data Sampler widget with a python program, and used the trainning data to train a Logistic Regression model(parameter is the same like previous) , used the test data to validate the model, but I got a different result both in trainning and testing(all the parameters are the same) compared with above results, why?
Another question is how to output the detail parameters when sampling (i.e. the random seed and the test data), is it feasible?
Maybe these questions are quite stupid, I hope to get your help, thanks!

**Comment by ajdapretnar** [comment]
If I understand correctly, you sampled both in Data Sampler and separately in the code. Sampling in this case is always random, this is why you get different results. If you want to get close to replicability, you can save the training and test data with Save Data. However, Logistic Regression is partially stochastic, too, so you can't reproduce it completely.
Random seed can't be saved, but test data can. Just connect Save Data to Data Sampler and set the output to Remaining Data.

**Comment by QiChen2014** [comment]
@ajdapretnar Thanks for your reply, I get it. I used the sampling result to invert the data index to determine the test set, now I know how to save the test data directly.  Some other questions: 
Is the data extraction completely random and there is no default random seed? When I chose “replicable sampling” and "stratify sample", the result of sampling is always the same. How did this happen?
Thanks again for your great work！
